helix-term/src/commands.rs

```diff
@@ -2326,10 +2326,17 @@ async fn make_format_callback(
         let view_id = view!(editor).id;
         if let Some(doc) = editor.document_mut(doc_id) {
             if doc.version() == doc_version {
-                doc.apply(&Transaction::from(format), view_id);
-                doc.append_changes_to_history(view_id);
-                if let Modified::SetUnmodified = modified {
-                    doc.reset_modified();
+                if let Some(text) = format.replaces_document(doc.text()) {
```

Comment:
I think we shouldn't do any heuristics here and instead just always re-detect line endings on formatting callbacks.

Reply:
I wouldn't call the same code as `reload()` and instead just use `self.detect_indent_and_line_ending();`

Reply:
changed by 0851f74e76c9031a4f0c9458a311bc0e6b96a7b3 to do it like that